src/utils/user-mapping.ts

```diff
@@ -0,0 +1,109 @@
+    const actingUser: AppActingUser | undefined = context.acting_user;
+    const i18nObj = configureI18n(context);
+
+    if (!actingUser) {
+        throw new Exception(ExceptionType.TEXT_ERROR, i18nObj.__('general.validation-user.not-provided'));
+    }
+
+    return isUserSystemAdmin(actingUser) || linkEmailAddress(oauth2);
+}
+
+export async function validateUserAccess(call: AppCallRequest): Promise<OpsUser> {
+    const actingUser: AppActingUser | undefined = call.context.acting_user;
+    const i18nObj = configureI18n(call.context);
+    const apiKey = getOpsGenieAPIKey(call);
+
+    if (!actingUser) {
+        throw new Exception(ExceptionType.TEXT_ERROR, i18nObj.__('general.validation-user.not-provided'));
+    }
+
+    const optionsOpsgenie: OpsGenieOptions = {
+        api_key: apiKey,
+    };
+    const opsGenieClient = new OpsGenieClient(optionsOpsgenie);
+
+    const identifierUser: Identifier = {
+        identifier: actingUser.email,
+        identifierType: IdentifierType.USERNAME,
+    };
+    const genieUser: OpsUser = await tryPromise<OpsUser>(opsGenieClient.getUser(identifierUser), ExceptionType.MARKDOWN, i18nObj.__('general.validation-user.genie-user', { email: actingUser.email }));
+
+    if (genieUser.blocked) {
+        throw new Exception(ExceptionType.TEXT_ERROR, i18nObj.__('general.validation-user.genie-user', { email: actingUser.email }));
+    }
+
+    if (!genieUser.verified) {
+        throw new Exception(ExceptionType.TEXT_ERROR, i18nObj.__('general.validation-user.genie-user-verified', { email: actingUser.email }));
+    }
+
+    return genieUser;
+}
+
+export async function canUserInteractWithAlert(call: AppCallRequest | AppCallAction<AppContextAction>, alertTinyId: string): Promise<Alert> {
+    const actingUser: AppActingUser | undefined = call.context.acting_user;
+    const isSystemAdmin: boolean = isUserSystemAdmin(actingUser);
+    const allowMember: boolean = allowMemberAction(call.context);
+    const i18nObj = configureI18n(call.context);
+    const apiKey = getOpsGenieAPIKey(call);
+
+    const optionsOpsgenie: OpsGenieOptions = {
+        api_key: apiKey,
+    };
+    const opsGenieClient = new OpsGenieClient(optionsOpsgenie);
+
+    const identifier: Identifier = {
+        identifier: alertTinyId,
+        identifierType: IdentifierType.TINY,
+    };
+
+    const alertResponse: Alert = await tryPromise<Alert>(opsGenieClient.getAlert(identifier), ExceptionType.MARKDOWN, i18nObj.__('forms.error'));
+
+    if (!allowMember) {
+        throw new Exception(ExceptionType.MARKDOWN, i18nObj.__('general.validation-user.genie-action-invalid'));
+    }
+
+    if (!isSystemAdmin) {
+        const teamIdentifier: Identifier = {
+            identifier: alertResponse.ownerTeamId,
+            identifierType: IdentifierType.ID,
+        };
+
+        const team: Team = await tryPromise<Team>(opsGenieClient.getTeam(teamIdentifier), ExceptionType.MARKDOWN, i18nObj.__('forms.error'));
```

Comment:
Is it possible we don't get all of the team members due to paging?

Reply:
On OpsGenie documentation [GET Team](https://docs.opsgenie.com/docs/team-api#get-team), there is nothing about team members being paginated